backstory/src/main/java/com/google/sps/servlets/data/BlobstoreManager.java

```diff
@@ -0,0 +1,140 @@
+
+/**
+ * Wrapper class for the blobstore service and all related operations.
+ */
+public class BlobstoreManager {
+  private BlobstoreService blobstoreService;
+  private BlobstoreServiceConstantFields blobstoreServiceConstantFields;
+  private BlobInfoFactory blobInfoFactory;
+
+  /**
+   * Creates a blobstore manager object by calling the blobstore service factory
+   */
+  public BlobstoreManager() throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    this.blobstoreService = blobstoreService;
+    this.blobstoreServiceConstantFields = new BlobstoreServiceConstantFields(blobstoreService);
+    this.blobInfoFactory = new BlobInfoFactory();
+  }
+
+  /**
+   * Injection code for testing.
+   * Creates a blobstore manager using a mock blobstoreService.
+   */
+  public BlobstoreManager(BlobstoreService blobstoreService,
+      BlobstoreServiceConstantFields blobstoreServiceConstantFields,
+      BlobInfoFactory blobInfoFactory) {
+    this.blobstoreService = blobstoreService;
+    this.blobstoreServiceConstantFields = blobstoreServiceConstantFields;
+    this.blobInfoFactory = blobInfoFactory;
+  }
+
+  /**
+   * Given a request and the form input element name get the image uploaded in the form as
+   * a Blob Key in String form. This key is used to serve the picture back to the front-end.
+   * @param request the HTTP request sent from the front-end form.
+   * @param formInputElementName the name of the input element in the front-end form.
+   * @return the image uploaded in the front-end input form, as a Blob Key in String form.
+   */
+  public String getUploadedFileBlobKeyString(
+      HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = blobInfoFactory.loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    String blobKeyString = blobKey.getKeyString();
+    return blobKeyString;
+  }
+
+  /**
+   * Given a request and the form input element name get the image uploaded in the form as
+   * a byte array.
+   * @param request the HTTP request sent from the front-end form.
+   * @param formInputElementName the name of the input element in the front-end form.
+   * @return the image uploaded in the front-end input form, as a byte array.
+   */
+  public byte[] getBlobBytes(HttpServletRequest request, String formInputElementName)
```

Comment:
There's some duplication between this method and `getUploadedFileBlobKeyString`. Is there any way to avoid that?